beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3676,34 +3668,72 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                     .start_of(head_slot)
                     .unwrap_or_else(|| Duration::from_secs(0)),
             );
+            let observed_delay = block_delays
+                .observed
+                .unwrap_or_else(|| Duration::from_secs(0));
+            let import_delay = block_delays
+                .imported
+                .unwrap_or_else(|| Duration::from_secs(0));
+            let attestable_delay = block_delays
+                .attestable
+                .unwrap_or_else(|| Duration::from_secs(0));
+            let set_as_head_delay = block_delays
+                .set_as_head
+                .unwrap_or_else(|| Duration::from_secs(0));
 
             metrics::observe_duration(
                 &metrics::BEACON_BLOCK_OBSERVED_SLOT_START_DELAY_TIME,
-                block_delays
-                    .observed
-                    .unwrap_or_else(|| Duration::from_secs(0)),
+                observed_delay,
+            );
+            metrics::observe_duration(
+                &metrics::BEACON_BLOCK_HEAD_ATTESTABLE_DELAY_TIME,
+                attestable_delay,
+            );
+            metrics::observe_duration(
+                &metrics::BEACON_BLOCK_IMPORTED_OBSERVED_DELAY_TIME,
+                import_delay,
             );
-
             metrics::observe_duration(
                 &metrics::BEACON_BLOCK_HEAD_IMPORTED_DELAY_TIME,
-                block_delays
-                    .set_as_head
-                    .unwrap_or_else(|| Duration::from_secs(0)),
+                set_as_head_delay,
             );
 
             // If the block was enshrined as head too late for attestations to be created for it,
             // log a debug warning and increment a metric.
-            if late_head {
-                metrics::inc_counter(&metrics::BEACON_BLOCK_HEAD_SLOT_START_DELAY_EXCEEDED_TOTAL);
+            let missed_attestation_deadline = attestable_delay >= attestation_deadline;
```

Comment:
Since this `if` statement is dependent on `attestable_delay`, and `attestable_delay` is only written to during the `early_attester` code path could we have cases where the block is not added to the `early_attester` cache while still being classed as `late`?

I'm not well versed with the inner workings of the `early_attester` cache, but I assume almost all head blocks run through it?

I'm envisioning a scenario where a block is very late, misses the early attester cache, therefore doesn't get the metrics below, or the debug log, but is still cached in `block_delays`, gets recorded in the above metrics and is also sent out via `SSE`.